python/pyiceberg/transforms.py

```diff
@@ -427,6 +486,20 @@ def can_transform(self, source: IcebergType) -> bool:
             TimestamptzType,
         }
 
+    def project(self, name: str, pred: BoundPredicate) -> Optional[UnboundPredicate]:
```

Comment:
Is this different than the other `DatesTransform` implementation? I might be missing it.